File: 0026-remove-duplicates-from-sorted-array/main.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int slow=0, fast =0;
        for(int n: nums){
            if(fast+1<nums.size() && nums[slow]!=nums[fast+1]){
                nums[++slow] = nums[fast+1];
            }
            fast++;
        }
        return slow+1;
    }
};

File: 0026-remove-duplicates-from-sorted-array/main.py
class Solution:
    def removeDuplicates(self, nums):
        if not nums:
            return 0

        slow = 0
        fast = 0

        while fast < len(nums) -1:
            if nums[slow] != nums[fast + 1]:
                slow += 1
                nums[slow] = nums[fast + 1]
            fast += 1

        return slow + 1

if __name__ == "__main__":
    sol = Solution()
    nums = [1, 1, 2, 2, 3]
    length = sol.removeDuplicates(nums)
    print("New length:", length)
    print("Array:", nums[:length])